src/clj/game/cards/ice.clj

```diff
@@ -2721,28 +2703,26 @@
                                            card nil)))}]})
 
 (defcard "Mlinzi"
-  (letfn [(net-or-trash [net-dmg mill-cnt]
+  (letfn [(net-or-mill [net-dmg mill-cnt]
             {:label (str "Do " net-dmg " net damage")
              :player :runner
              :waiting-prompt true
              :prompt "Choose one"
              :choices (req [(str "Take " net-dmg " net damage")
-                            (when (<= mill-cnt (count (:deck runner)))
-                              (str "Trash the top " (quantify mill-cnt "card") " of the stack"))])
+                            (when (can-pay? state :runner eid card nil [:trash-from-deck mill-cnt])
+                              (capitalize (build-cost-label [:trash-from-deck mill-cnt])))])
              :async true
              :effect (req (if (= target (str "Take " net-dmg " net damage"))
                             (do (system-msg state :corp
                                             (str "uses Mlinzi to do "
                                                  net-dmg " net damage"))
                                 (damage state :runner eid :net net-dmg {:card card}))
-                            (do (system-msg state :corp
-                                            (str "uses Mlinzi to trash "
-                                                 (enumerate-str (map :title (take mill-cnt (:deck runner))))
-                                                 " from the runner's stack"))
-                                (mill state :runner eid :runner mill-cnt))))})]
-    {:subroutines [(net-or-trash 1 2)
-                   (net-or-trash 2 3)
-                   (net-or-trash 3 4)]}))
+                            (wait-for (pay state :runner (make-eid state eid) card [:trash-from-deck mill-cnt])
```

Comment:
> Do 1 net damage unless the Runner trashes the top 2 cards of the stack.

This is a cost?

Reply:
Yes in the sense that if the runner must pay it or else suffer the net damage. As in Turing, Public Trail, or Data Ward the main effect can be avoided by the Runner if they pay an alternative cost, which in the case of Formicary is the 2 net damage. The Runner cannot prevent this damage or else Formicary would end the run, the same that occurs with Obokata Protocol's steal cost.

Reply:
Cool thanks